python/lsst/cp/pipe/cpFilterScan.py

```diff
@@ -0,0 +1,377 @@
+
+        filterScan = []
+        for wavelength in filterScanResults.keys():
+            scans = filterScanResults[wavelength]
+
+            referenceScan = [x for x in scans if x['physical_filter'] == self.config.referenceFilter]
+            if len(referenceScan) == 0:
+                # No reference scan at this wavelength.
+                continue
+            referenceScan = referenceScan[0]
+            referenceValue = referenceScan['scale'] / referenceScan['flux']
+
+            wavelengthScan = {'wavelength': float(wavelength)}
+            for filterName in filterSet:
+                wavelengthScan[filterName] = np.nan
+
+            for scan in scans:
+                if np.isfinite(wavelengthScan[scan['physical_filter']]):
+                    self.log.warn(f"Multiple instances of filter: {scan['physical_filter']}")
+                wavelengthScan[scan['physical_filter']] = referenceValue * scan['flux'] / scan['scale']
+
+            filterScan.append(copy.copy(wavelengthScan))
+
+        outMetadata = PropertyList()
+        self.combine.combineHeaders(inputExpHandles, calib=None, metadata=outMetadata,
+                                    calibType="FilterScan", scales=None)
+        filteredMetadata = {k: v for k, v in outMetadata.toDict().items() if v is not None}
+
+        catalog = Table(filterScan)
+        catalog.meta = filteredMetadata
+        efdClient.close()
+        return pipeBase.Struct(outputData=catalog)
+
+
+class CpMonochromatorScanConnections(pipeBase.PipelineTaskConnections,
+                                     dimensions=("instrument", )):
+    inputExpHandles = cT.Input(
+        name="rawSpectrum",
+        doc="Input spectrograph measurements.",
+        storageClass="FiberSpectrum",
+        dimensions=("instrument", "exposure", "detector"),
+        multiple=True,
+        deferLoad=True,
+    )
+    outputData = cT.Output(
+        name="cpMonochromatorScan",
+        doc="Output table to write.",
+        storageClass="ArrowAstropy",
+        dimensions=("instrument", ),
+    )
+
+
+class CpMonochromatorScanConfig(pipeBase.PipelineTaskConfig,
+                                pipelineConnections=CpMonochromatorScanConnections):
+    efdClientInstance = pexConfig.Field(
+        dtype=str,
+        doc="EFD instance to use for monochromator results.",
+        default="usdf_efd",
+    )
+    efdUnits = pexConfig.Field(
+        dtype=str,
+        doc="Units to use for EFD monochromator results.",
+        default="nm",
+    )
+    headerDateKey = pexConfig.Field(
+        dtype=str,
+        doc="Header keyword to use for observation date.",
+        default="DATE-BEG",
+    )
+    peakBoxSize = pexConfig.Field(
+        dtype=int,
+        doc="Half-size of the box used for fitting the spectrum peak.",
+        default=50,
+    )
+
+
+class CpMonochromatorScanTask(pipeBase.PipelineTask):
+    """Compare EFD monochromator results to fiber spectrograph spectra.
+    """
```

Comment:
Is this used in any of the pipelines?  I don't know what this task is.

Reply:
Documentation updated as mentioned above.  This is used in an `_ingredients/cpMonochromatorScan.yaml`, which I did not add a camera specific task for, as we currently only have one fiber spectrograph (and so the config is specific to that instrument).